dot/parachain/backing/per_relay_parent_state.go

```diff
@@ -416,3 +423,35 @@ func executorParamsAtRelayParent(
 	// https://github.com/paritytech/polkadot-sdk/blob/7ca0d65f19497ac1c3c7ad6315f1a0acb2ca32f8/polkadot/node/subsystem-util/src/lib.rs#L241-L242
 	return parachaintypes.ExecutorParams{}, nil
 }
+
+func getPovFromValidator(
+	subSystemToOverseer chan<- any,
+	chRelayParentAndCommand chan relayParentAndCommand,
+	relayParent common.Hash,
+	candidateHash parachaintypes.CandidateHash,
+	attesting *attestingData,
+) (*parachaintypes.PoV, error) {
+	fetchPov := parachaintypes.AvailabilityDistributionMessageFetchPoV{
+		RelayParent:   relayParent,
+		FromValidator: attesting.fromValidator,
+		ParaID:        parachaintypes.ParaID(attesting.candidate.Descriptor.ParaID),
+		CandidateHash: candidateHash,
+		PovHash:       attesting.povHash,
+		PovCh:         make(chan parachaintypes.OverseerFuncRes[parachaintypes.PoV]),
+	}
+
+	subSystemToOverseer <- fetchPov
+	PovRes := <-fetchPov.PovCh
```

Comment:
This way out program will get stuck until we receive a response in the channel. If for some problem, response never gets sent we will be left waiting forever.

Let's put a timeout here. If we don't get a response in the required time, throw an error
You can use the code from here
https://github.com/ChainSafe/gossamer/blob/f8ddc007c496d5e1e97dffd796da337bc90e2eb5/dot/parachain/collator-protocol/message.go#L209-L215